app/analyzers/sudden_appearance.py

```diff
@@ -0,0 +1,144 @@
+        self.model_settings["target"] in events within the time window defined by start_slide_win and en_slide_win
+        and create outliers. An event is considered as outlier when a term field appear for the first time after
+        a certain proportion of the time window. This proportion is defined by
+        self.model_settings["trigger_slide_window_proportion"].
+
+        :param start_slide_win: start time of the time window
+        :param end_slide_win: end time of the time window
+        """
+        aggregation_buckets = es.scan_first_occur_documents(search_query=self.search_query,
+                                                            start_time=start_slide_win,
+                                                            end_time=end_slide_win,
+                                                            model_settings=self.model_settings)
+        # Loop over the aggregations
+        for bucket in aggregation_buckets:
+            bucket_docs = bucket["target"]["buckets"]
+            # Loop over the documents in aggregation
+            for doc in bucket_docs:
+                raw_doc = doc["top_doc"]["hits"]["hits"][0]
+                fields = es.extract_fields_from_document(raw_doc,
+                                                         extract_derived_fields=self.model_settings[
+                                                             "use_derived_fields"])
+                # convert the event timestamp in the right format
+                event_timestamp = dateutil.parser.parse(fields[self.model_settings["timestamp_field"]],
+                                                        ignoretz=True)
+                # Compute the proportion, compared to the time window, the event appear
+                prop_time_win_event_appears = (event_timestamp - start_slide_win) / (end_slide_win - start_slide_win)
+                if prop_time_win_event_appears > self.model_settings["trigger_slide_window_proportion"]:
+                    # retrieve extra information
+                    extra_outlier_information = dict()
+                    extra_outlier_information["prop_first_appear_in_time_window"] = prop_time_win_event_appears
+                    trigger_slide_window_proportion_str = str(self.model_settings["trigger_slide_window_proportion"])
+                    extra_outlier_information["trigger_slide_window_proportion"] = trigger_slide_window_proportion_str
+                    extra_outlier_information["size_time_window"] = str(self.delta_slide_win)
+                    extra_outlier_information["start_time_window"] = str(start_slide_win)
+                    extra_outlier_information["end_time_window"] = str(end_slide_win)
+                    extra_outlier_information["aggregator"] = self.model_settings["aggregator"]
+                    extra_outlier_information["aggregator_value"] = bucket["key"]
```

Comment:
Avoid having multiple names for the same concept (aggregator and bucket key for example)